utils/s2n_safety.c

```diff
@@ -90,12 +90,21 @@ int s2n_constant_time_copy_or_dont(uint8_t * dest, const uint8_t * src, uint32_t
     S2N_PUBLIC_INPUT(src);
     S2N_PUBLIC_INPUT(len);
 
+/* This underflows a value of 0 to the maximum value via arithmetic underflow,
+ * so the check for arithmetic overflow/underflow needs to be disabled for CBMC.
+ * Additionally, uint_fast16_t is defined as the fastest available unsigned
+ * integer with 16 bits or greater, and is not guaranteed to be 16 bits long.
+ * To handle this, the conversion overflow check also needs to be enabled. */
+#pragma CPROVER check push
+#pragma CPROVER check disable "conversion"
+#pragma CPROVER check disable "unsigned-overflow"
     uint8_t mask = ((uint_fast16_t)((uint_fast16_t)(dont) - 1)) >> 8;
+#pragma CPROVER check pop
```

Comment:
@camshaft any thoughts here? Any suggestions that could prevent us from using the pragmas?

Reply:
if `dont` is 0, we don't need to perform this operation, right? That would avoid the possible underflow.

Reply:
could you create an issue for this one too?